nodejs/nodejs-guestbook/skaffold.yaml

```diff
@@ -1,43 +1,27 @@
-# To learn more about the skaffold.yaml schema visit
-# https://skaffold.dev/docs/references/yaml/
-apiVersion: skaffold/v2beta4
+# The Guestbook app is configured to use skaffold modules.
+# The frontend and backend services can be activated by name,
+# allowing each service to be iterated on independently.
+# To learn more about skaffold modules and multi-config support,
+# visit https://skaffold.dev/docs/design/config/
+
+apiVersion: skaffold/v2beta15
 kind: Config
+metadata:
+  name: frontend-svc
 build:
-  tagPolicy:
-    sha256: {}
-  # defines where to find the code at build time and where to push the resulting image
-  artifacts:
-  - image: nodejs-guestbook-backend
-    context: src/backend
-  - image: nodejs-guestbook-frontend
-    context: src/frontend
-# defines the Kubernetes manifests to deploy on each run
-deploy:
-  kubectl:
-    manifests:
-    - ./kubernetes-manifests/*.service.yaml
-    - ./kubernetes-manifests/*.deployment.yaml
-portForward:
-  # Cloud Code automatically port forwards resources,
-  # or users can define explicit port forwards here.
-  # For more info, visit https://skaffold.dev/docs/pipeline-stages/port-forwarding/#UDPF
-  - resourceType: service
-    resourceName: nodejs-guestbook-frontend
-    port: 80
-    localPort: 4503
-profiles:
-# use the cloudbuild profile to build images using Google Cloud Build
-- name: cloudbuild
-  build:
-    googleCloudBuild: {}
-- name: buildpacks
-  build:
-    artifacts:
-    - image: nodejs-guestbook-backend
-      context: src/backend
-      buildpack:
-        builder: "gcr.io/buildpacks/builder:v1"
-    - image: nodejs-guestbook-frontend
-      context: src/frontend
-      buildpack:
-        builder: "gcr.io/buildpacks/builder:v1"
+  local:
+    concurrency: 0
+requires:
+  - path: ./src/frontend
+
+---
```

Comment:
I see the preference of having multiple modules (or unnamed configurations too) in all our existing module samples - `bank-of-anthos`, `multi-config-microservices`, etc. This seems to be sort of a new trend given we almost never used the multiple documents in a single YAML file before. Are the modules especially suited for multiple document approach? @gsquared94 @briandealwis

Reply:
It's a coupling vs cohesiveness tradeoff, IMHO.  Are the modules (or at least one of them) independent?  Then have separate files.